File: backend/app/schemas/tender.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel, ConfigDict


class TenderBase(BaseModel):
    reference_number: str
    title: str
    entity: Optional[str] = None
    activity: Optional[str] = None
    tender_type: Optional[str] = None
    tender_number: Optional[str] = None
    publish_date: Optional[datetime] = None
    last_enquiry_date: Optional[datetime] = None
    last_offer_date: Optional[datetime] = None
    award_date: Optional[datetime] = None
    document_price: Optional[float] = None
    document_price_currency: Optional[str] = None
    tender_details: Optional[str] = None
    purpose: Optional[str] = None
    conditions: Optional[str] = None
    location: Optional[str] = None
    quantity: Optional[str] = None
    details_url: Optional[str] = None


class TenderCreate(TenderBase):
    raw_html: Optional[str] = None
    scraped_at: Optional[datetime] = None


class TenderRead(TenderBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    is_relevant: Optional[bool] = None
    relevance_score: Optional[float] = None
    relevance_reason: Optional[str] = None
    matched_at: Optional[datetime] = None
    notification_sent: bool
    notification_sent_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime
    scraped_at: Optional[datetime] = None


class TenderList(BaseModel):
    items: List[TenderRead]
    total: int
    page: int
    page_size: int
    pages: int


class ScrapeJobRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    status: str
    started_at: Optional[datetime] = None
    finished_at: Optional[datetime] = None
    tenders_found: int
    tenders_new: int
    pages_scraped: int
    error_message: Optional[str] = None
    created_at: datetime


class TenderFilter(BaseModel):
    search: Optional[str] = None
    is_relevant: Optional[bool] = None
    notification_sent: Optional[bool] = None
    entity: Optional[str] = None
    activity: Optional[str] = None
    page: int = 1
    page_size: int = 20

File: frontend/src/types/tender.ts
export interface Tender {
  id: number;
  reference_number: string;
  title: string;
  entity: string | null;
  activity: string | null;
  tender_type: string | null;
  publish_date: string | null;
  last_enquiry_date: string | null;
  last_offer_date: string | null;
  award_date: string | null;
  document_price: number | null;
  document_price_currency: string | null;
  tender_details: string | null;
  purpose: string | null;
  conditions: string | null;
  location: string | null;
  details_url: string | null;
  is_relevant: boolean | null;
  relevance_score: number | null;
  relevance_reason: string | null;
  matched_at: string | null;
  notification_sent: boolean;
  notification_sent_at: string | null;
  created_at: string;
  updated_at: string;
  scraped_at: string | null;
}

export interface TenderList {
  items: Tender[];
  total: number;
  page: number;
  page_size: number;
  pages: number;
}

export interface TenderFilters {
  search?: string;
  is_relevant?: boolean;
  notification_sent?: boolean;
  entity?: string;
  activity?: string;
  page: number;
  page_size: number;
}
